ssdp.md

```diff
@@ -0,0 +1,135 @@
+    PRESENTATION-ENDPOINT.example.org: 192.168.1.100:3000
+    SUPPORTED-URLS.example.org: http://example.com/foo.html
+    ```
+* The display sends the following SSDP message when the receiver service in not more available. There are no new headers used in the byebye message.
+
+    ```
+    NOTIFY * HTTP/1.1
+    HOST: 239.255.255.250:1900
+    NT: urn:example-org:service:osp:1
+    NTS: ssdp:byebye
+    USN: uuid:advertisement UUID
+    ```
+
+### Remote Playback API: Monitor Display Availability
+TODO (it should be very similar to the Presentation API display availability but for capabilities other parameters related media formats, codecs, encryptions need to be considered)
+
+## Non-functional Requirements
+
+### Reliability
+
+UPnP recommend due to the unreliable nature of UDP to send the SSDP messages more than once (and max three times to avoid network congestion) with some delay of few hundred milliseconds. In addition, the device must re-send its advertisements periodically prior to expiration of the duration specified in
+the `CACHE-CONTROL` header (minimum value is 1800s).  
+
+### Latency of device discovery / device removal
+
+New devices added or removed can be immediately detected if the controller listens to the multicast address for "ssdp:alive" and "ssdp:byebye" messages. For search requests the latency depends on the `MX` SSDP header which contains the maximum wait time in seconds. According to UPnP specification, it must be greater than or equal to 1 and should be less than 5 inclusive. SSDP responses should be delayed a random duration between 0 and the value of `MX` to balance load for the controller when it processes responses.
+
+### Ease of implementation / deployment
+
+It is very easy to implement the SSDP protocol. It is based on UDP and the messages are easy to create and parse. [peer-ssdp](https://github.com/fraunhoferfokus/peer-ssdp) is a open source implementation for Node.js by Fraunhofer FOKUS and shows how easy to implement the protocol. Another implementation by Fraunhofer FOKUS for Android (part of a [cordova plugin](https://github.com/fraunhoferfokus/cordova-plugin-hbbtv/tree/master/src/android/ssdp)) is also available. 
+
+### Security - both of the implementation, and whether it can be leveraged to enhance security of the entire protocol
+
+SSDP consider local network as secure environment. Any device on the network can discover other devices or services. Security can be implemented on service level. In our case the output of the discovery is a list of displays that contain friendly names and endpoints of presentation receiver services. Any security mechanism can be implemented during session establishment and communication independent from the method used for discovery. 
+
+### Privacy: what device information is exposed
+
+The standard UPnP device description exposes parameters about the device/service like unique identifier, friendly name, software, manufacturer, service endpoints, etc.
```

Comment:
I think it is very important to clearly define what is the input and output of the discovery process. Exposing friendly name reduce the number of calls needed to show the Display Selection Dialog on the client. UUID helps in case of SSDP to detect device removal or detect messages from same device.